src/data/matrix/kronecker.lean

```diff
@@ -153,6 +154,13 @@ begin
   simp_rw [f.map_sum, linear_map.sum_apply, linear_map.map_sum, h_comm],
 end
 
+lemma kronecker_map_assoc {δ ξ ω ω' : Type*} (f : α → β → γ) (g : γ → δ → ω) (f' : α → ξ → ω')
+  (g' : β → δ → ξ) (A : matrix l m α) (B : matrix n p β) (D : matrix q r δ) (φ : ω ≃ ω')
+  (hφ : ∀ a b d, φ (g (f a b) d ) = f' a (g' b d )) :
+  (reindex (equiv.prod_assoc l n q) (equiv.prod_assoc m p r)).trans (equiv.map_matrix φ)
+    (kronecker_map g (kronecker_map f A B) D) = kronecker_map f' A (kronecker_map g' B D) :=
+ext $ λ i j, hφ _ _ _
```

Comment:
If you don't want to remove `φ`, it might make sense to relax it to be any function:
```suggestion
lemma kronecker_map_assoc {δ ξ ω ω' : Type*} (f : α → β → γ) (g : γ → δ → ω) (f' : α → ξ → ω')
  (g' : β → δ → ξ) (A : matrix l m α) (B : matrix n p β) (D : matrix q r δ) (φ : ω → ω')
  (hφ : ∀ a b d, φ (g (f a b) d ) = f' a (g' b d )) :
  reindex (equiv.prod_assoc l n q) (equiv.prod_assoc m p r)
    ((kronecker_map g (kronecker_map f A B) D).map φ) = kronecker_map f' A (kronecker_map g' B D) :=
```
Another option might just be to remove this proof in its entirety, as the name is longer than the proof.

I'll leave it to @jcommelin or @Vierkantor to make the final call.